qiskit/quantum_info/operators/symplectic/clifford_circuits.py

```diff
@@ -39,10 +39,44 @@ def _append_circuit(clifford, circuit, qargs=None):
     if qargs is None:
         qargs = list(range(clifford.num_qubits))
 
+    if isinstance(circuit, Instruction):
+        return _append_operation(clifford, circuit, qargs)
     if isinstance(circuit, QuantumCircuit):
-        gate = circuit.to_instruction()
-    else:
-        gate = circuit
+        qubit_indices = {bit: idx for idx, bit in enumerate(circuit.qubits)}
+        for instruction in circuit:
+            if instruction.clbits:
+                raise QiskitError(
+                    f"Cannot apply Instruction with classical bits: {instruction.operation.name}"
+                )
+            # Get the integer position of the flat register
+            new_qubits = [qargs[qubit_indices[tup]] for tup in instruction.qubits]
+            _append_operation(clifford, instruction.operation, new_qubits)
+        return clifford
+
+    raise QiskitError("The circuit must be QuantumCircuit or Instruction")
```

Comment:
+1. Validation, if needed, should have happened at an earlier point. Usually `assert` is preferred if you want to catch a programming error, say for debugging, rather than a user error, that is data validation. I don't know where to find a succinct, authoritative explanation, but there are many sources online, like [this one](https://realpython.com/python-assert-statement/#when-not-to-use-assertions) that tell part of the story.

Reply:
Fixed in [349390d](https://github.com/Qiskit/qiskit-terra/pull/8585/commits/349390d490e1691746910ee669bdbba5969a9580)